hypothesis-python/src/hypothesis/internal/conjecture/datatree.py

```diff
@@ -67,6 +61,59 @@ class Conclusion:
     interesting_origin = attr.ib()
 
 
+def compute_max_children(kwargs, ir_type):
+    if ir_type == "integer":
+        min_value = kwargs["min_value"]
+        max_value = kwargs["max_value"]
+        if min_value is not None and max_value is not None:
+            return max_value - min_value + 1
+        if min_value is not None:
+            return (2**127) - min_value
+        if max_value is not None:
+            return (2**127) + max_value
+        return 2**128 - 1
+    elif ir_type == "boolean":
+        return 2
+    elif ir_type == "bytes":
+        return 2 ** (8 * kwargs["size"])
+    elif ir_type == "string":
+        min_size = kwargs["min_size"]
+        max_size = kwargs["max_size"]
+        intervals = kwargs["intervals"]
+
+        if max_size is None:
+            # TODO extract this magic value out now that it's used in two places.
+            max_size = 10**10
+
+        # special cases for empty string, which has a single possibility.
+        if min_size == 0 and max_size == 0:
+            return 1
+
+        count = 0
+        if min_size == 0:
+            # empty string case.
+            count += 1
+            min_size = 1
+
+        count += len(intervals) ** (max_size - min_size + 1)
+        return count
+    elif ir_type == "float":
+        import ctypes
+
+        # number of 64 bit floating point values in [a, b].
+        # TODO this is wrong for [-0.0, +0.0], or anything that crosses the 0.0
+        # boundary.
+        # it also seems maybe wrong for [0, math.inf] but I haven't verified.
+        def binary(num):
+            # rely on the fact that adjacent floating point numbers are adjacent
+            # in their bitwise representation (except for -0.0 and +0.0).
+            return ctypes.c_uint64.from_buffer(ctypes.c_double(num)).value
```

Comment:
FYI: elsewhere I believe we have some similar logic which uses the `struct` module, unsure whether there's reusable code though.

Reply:
it turns out that not only is `float_to_int` equivalent to this `binary` function, but `count_between_floats` already exists! I don't know how I missed that.